src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs

```diff
@@ -570,95 +552,6 @@ private void EmitReversePInvokesAndShadowStackBottom()
             }
         }
 
-        private void EmitNativeMain(LLVMContextRef context)
-        {
-            LLVMValueRef shadowStackTop = Module.GetNamedGlobal("t_pShadowStackTop");
-
-            LLVMBuilderRef builder = context.CreateBuilder();
-            var mainSignature = LLVMTypeRef.CreateFunction(LLVMTypeRef.Int32, new LLVMTypeRef[] { LLVMTypeRef.Int32, LLVMTypeRef.CreatePointer(LLVMTypeRef.Int8, 0) }, false);
-            var mainFunc = Module.AddFunction("__managed__Main", mainSignature);
-            var mainEntryBlock = mainFunc.AppendBasicBlock("entry");
-            builder.PositionAtEnd(mainEntryBlock);
-            LLVMValueRef managedMain = Module.GetNamedFunction("StartupCodeMain");
-            if (managedMain.Handle == IntPtr.Zero)
-            {
-                throw new Exception("Main not found");
-            }
-
-            LLVMTypeRef reversePInvokeFrameType = LLVMTypeRef.CreateStruct(new LLVMTypeRef[] { LLVMTypeRef.CreatePointer(LLVMTypeRef.Int8, 0), LLVMTypeRef.CreatePointer(LLVMTypeRef.Int8, 0) }, false);
-            LLVMValueRef reversePinvokeFrame = builder.BuildAlloca(reversePInvokeFrameType, "ReversePInvokeFrame");
-            LLVMValueRef rhpReversePInvoke = Module.GetNamedFunction("RhpReversePInvoke");
-
-            builder.BuildCall(rhpReversePInvoke, new LLVMValueRef[] { reversePinvokeFrame }, "");
-
-            var shadowStack = builder.BuildMalloc(LLVMTypeRef.CreateArray(LLVMTypeRef.Int8, 1000000), String.Empty);
-            var castShadowStack = builder.BuildPointerCast(shadowStack, LLVMTypeRef.CreatePointer(LLVMTypeRef.Int8, 0), String.Empty);
-            builder.BuildStore(castShadowStack, shadowStackTop);
-
-            var shadowStackBottom = Module.GetNamedGlobal("t_pShadowStackBottom");
-            builder.BuildStore(castShadowStack, shadowStackBottom);
-
-            // Pass on main arguments
-            LLVMValueRef argc = mainFunc.GetParam(0);
-            LLVMValueRef argv = mainFunc.GetParam(1);
-
-            LLVMValueRef mainReturn = builder.BuildCall(managedMain, new LLVMValueRef[]
-            {
-                castShadowStack,
-                argc,
-                argv,
-            },
-            "returnValue");
-
-            LLVMValueRef rhpReversePInvokeReturn = Module.GetNamedFunction("RhpReversePInvokeReturn");
-            LLVMTypeRef reversePInvokeFunctionType = LLVMTypeRef.CreateFunction(LLVMTypeRef.Void, new LLVMTypeRef[] { LLVMTypeRef.CreatePointer(reversePInvokeFrameType, 0) }, false);
-            builder.BuildCall(rhpReversePInvokeReturn, new LLVMValueRef[] { reversePinvokeFrame }, "");
-
-            builder.BuildRet(mainReturn);
-            mainFunc.Linkage = LLVMLinkage.LLVMExternalLinkage;
-        }
-
-        private void EmitNativeStartup(LLVMContextRef context)
```

Comment:
I dont think we have a test for this do we?

Reply:
Indeed we do not (but should).

I have tested it manually and it works.

Notably, https://stackoverflow.com/a/70705808 needs to be updated to account for the fact `CoreRT_StaticInitialization` was renamed into `NativeAOT_StaticInitialization`.